doc/api-management.md

```diff
@@ -0,0 +1,118 @@
+# API Management
+
+> Guidance below for Theia framework. End products will get better development experience by using private visibility.
+
+- [**Stability**](#stability)
+  - [**Experimental**](#experimental)
+  - [**Stable**](#stable)
+- [**Finalization**](#finalization)
+- [**Deprecation**](#deprecation)
+
+Theia is a framework embracing openness, extensibility, and customizability as much as possible:
+- API defaults to public for clients.
+- API defaults to protected for extenders.
+- Language constructions prohibiting runtime access to internals are never used and so on.
+
+Usually, the version management is built around API visibility.
+Particularly, if a public API is broken then the major release is required.
+
+Following the conventional approach is not practicable for Theia since all APIs are more or less public.
+It will slow down API innovation, accumulate technical debt or require many major releases.
+
+Instead, we require a major release only if a stable API is broken.
+Breaking an experimental API can be done with a minor release.
+
+## Stability
+
+Conceptually, an API is all assumptions adopters depend on to make use or extend Theia.
+A stable API is based on assumptions that are not subject to change.
+It does not matter whether an API is public or internal, or whether it is used a lot or never.
+
+> For instance, the language server protocol depends on such data types like URIs and positions
+since they don't change from language to language. It ensures its stability.
+
+API stability is indicated explicitly by adding `@experimental` or `@stable` js-doc tags.
+An API without a stability tag is considered to be experimental.
+
+The explicit stability tag should mention since which version an API is stable or experimental.
+
+```ts
+/**
+ * @since 0.1.0 - added in 0.1.0
+ * @stable since 1.0.0 - finalized in 1.0.0
```

Comment:
`@stable since 1.0.0` and `finalized in 1.0.0` are redundant. Do we need any more explanation than `@stable since 1.0.0`?

Reply:
I will remove it, wanted to explicit